pkgs/tools/misc/nncp/default.nix

```diff
@@ -0,0 +1,55 @@
+{ stdenv, go, fetchurl, curl, perl, genericUpdater, writeShellScript }:
+
+stdenv.mkDerivation rec {
+  pname = "nncp";
+  version = "5.3.3";
+
+  buildInputs = [ go ];
+
+  preConfigure = ''
+    export GOCACHE=$PWD/.cache
+  '';
+
+  makeFlags = [
+    "PREFIX=${placeholder "out"}"
+    "CFGPATH=/etc/nncp.hjson"
+    "SENDMAIL=/run/wrappers/bin/sendmail"
+  ];
+
+  src = fetchurl {
+    url = "http://www.nncpgo.org/download/${pname}-${version}.tar.xz";
+    sha256 = "1l35ndzrvpfim29jn1p0bwmc8w892z44nsrdnay28k229r9dhz3h";
+  };
+
+  passthru.updateScript = genericUpdater {
+    inherit pname version;
+    versionLister = writeShellScript "nncp-versionLister" ''
+      echo "# Versions for $1:" >> "$2"
+      ${curl}/bin/curl -s http://www.nncpgo.org/Tarballs.html | ${perl}/bin/perl -lne 'print $1 if /Release.*>([0-9.]+)</'
```

Comment:
You may find this more elegant:

```
${curl}/bin/curl -s http://www.nncpgo.org/Tarballs.html | ${pup}/bin/pup 'body > table:nth-child(9) > tbody:nth-child(2) > tr:nth-child(1) > td:nth-child(1) > a:nth-child(1) text{}'
```

Reply:
nice, I did not know pup. Will try to explore the pup expression and stay in perl first.